fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/StringArithmetic.java

```diff
@@ -1010,6 +1004,29 @@ public static Expression quote(StringLikeLiteral first) {
      */
     @ExecFunction(name = "replace_empty")
     public static Expression replaceEmpty(StringLikeLiteral first, StringLikeLiteral second, StringLikeLiteral third) {
+        if (second.getValue().isEmpty()) {
+            if (first.getValue().isEmpty()) {
+                return castStringLikeLiteral(first, third.getValue());
+            }
+            byte[] input = first.getValue().getBytes(StandardCharsets.UTF_8);
+            byte[] replace = third.getValue().getBytes(StandardCharsets.UTF_8);
+            byte[] output = new byte[input.length + (input.length + 1) * replace.length];
+            for (int i = 0; i < replace.length; i++) {
+                output[i] = replace[i];
+            }
+            int last = replace.length;
+            for (int i = 0; i < input.length; i++) {
+                for (int j = 0; j < input.length; j++) {
+                    output[last] = input[j];
+                    last++;
+                }
+                for (int j = 0; j < replace.length; j++) {
+                    output[last] = replace[j];
+                    last++;
+                }
+            }
```

Comment:
maybe reuse splitByGrapheme is a better way. let splitByGrapheme return List<String> instead of ArrayLiteral. and then we can use it here to split input